lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -38,7 +38,7 @@
     }
 
     while (k < len) {
-        if (o[k] === searchElement && (o[k] !== undefined || searchElement !== undefined)) {
+        if (o[k] === searchElement && (o[k] !== undefined || (searchElement === undefined && o.hasOwnProperty(k)))) {
```

Comment:
Sooo, I changed `k in o` for performances reason. If you remember, with this expression we had a regression of 20% on Splay. As @dilijev said, that weirdness is only true if the indexes are outside the length of the array or are non-integer numbers. So the behavior is similar to `k in o` and is fast, except in some improbable weird case like searching for undefined in an almost empty array where the length is defined manually.